packages/tldraw/src/lib/shapes/arrow/elbow/routes/elbowArrowRoutes.tsx

```diff
@@ -0,0 +1,432 @@
+}
+
+/**
+ * See `routeRightToTop`.
+ */
+export function routeRightToBottom(info: ElbowArrowWorkingInfo): ElbowArrowRoute | null {
+	info.apply(ElbowArrowTransform.FlipY)
+	return routeRightToTop(info)
+}
+
+/**
+ * Arrows may be mirrored - Y flipped
+ * ```
+ * 1:        2:                3:
+ * ┌───┐     ┌───┐ ┌───────┐           ┌───┐
+ * │ A ├─┐   │ A ├─┘ ┌───┐ │   ┌───┐   │ A ├─┐
+ * └───┘ │   └───┘   │ B ◄─┘   │ B ◄─┐ └───┘ │
+ * ┌───┐ │           └───┘     └───┘ └───────┘
+ * │ B ◄─┘
+ * └───┘
+ * ```
+ */
+export function routeRightToRight(info: ElbowArrowWorkingInfo): ElbowArrowRoute | null {
+	const aEdge = info.A.edges.right
+	const bEdge = info.B.edges.right
+
+	if (!aEdge || !bEdge) return null
+
+	if (
+		(info.gapX < 0 ||
+			aEdge.crossTarget > info.B.expanded.bottom ||
+			aEdge.crossTarget < info.B.expanded.top) &&
+		(bEdge.value > info.A.original.left ||
+			bEdge.crossTarget > info.A.expanded.bottom ||
+			bEdge.crossTarget < info.A.expanded.top)
+	) {
+		// Arrow 1
+		return new ElbowArrowRouteBuilder(info, 'to right 1')
+			.add(aEdge.value, aEdge.crossTarget)
+			.add(info.common.expanded.right, aEdge.crossTarget)
+			.add(info.common.expanded.right, bEdge.crossTarget)
+			.add(bEdge.value, bEdge.crossTarget)
+			.build()
+	}
+
+	if (info.midX === null) return null
+
+	if (bEdge.expanded !== null && info.gapX > 0) {
+		const viaBottomLength =
+			Math.abs(bEdge.crossTarget - info.B.expanded.bottom) +
+			Math.abs(aEdge.crossTarget - info.B.expanded.bottom)
+		const viaTopLength =
+			Math.abs(bEdge.crossTarget - info.B.expanded.top) +
+			Math.abs(aEdge.crossTarget - info.B.expanded.top)
+
+		const topOrBottom = viaBottomLength < viaTopLength ? 'bottom' : 'top'
+		// Arrow 2:
+		return new ElbowArrowRouteBuilder(info, `to right 2 via ${topOrBottom}`)
+			.add(aEdge.value, aEdge.crossTarget)
+			.add(info.midX, aEdge.crossTarget)
+			.add(info.midX, info.B.expanded[topOrBottom])
+			.midpointHandle('x')
+			.add(bEdge.expanded, info.B.expanded[topOrBottom])
+			.add(bEdge.expanded, bEdge.crossTarget)
+			.add(bEdge.value, bEdge.crossTarget)
+			.build()
+	}
+
+	if (aEdge.expanded !== null && info.gapX < 0) {
+		const viaBottomLength =
+			Math.abs(bEdge.crossTarget - info.A.expanded.bottom) +
+			Math.abs(aEdge.crossTarget - info.A.expanded.bottom)
+		const viaTopLength =
+			Math.abs(bEdge.crossTarget - info.A.expanded.top) +
+			Math.abs(aEdge.crossTarget - info.A.expanded.top)
+
+		const topOrBottom = viaBottomLength < viaTopLength ? 'bottom' : 'top'
+		// Arrow 3:
+		return new ElbowArrowRouteBuilder(info, `to right 3 via ${topOrBottom}`)
+			.add(aEdge.value, aEdge.crossTarget)
+			.add(aEdge.expanded, aEdge.crossTarget)
+			.add(aEdge.expanded, info.A.expanded[topOrBottom])
+			.add(info.midX, info.A.expanded[topOrBottom])
+			.add(info.midX, bEdge.crossTarget)
+			.midpointHandle('x')
+			.add(bEdge.value, bEdge.crossTarget)
+			.build()
+	}
+
+	return null
+}
+
+const routes = {
+	top: {
+		top: (info) => {
```

Comment:
wdyt about making this a terser format?
`top: [Rotate270, routeRightToRight]`
and then we interpret it later.

Reply:
could do? i'd rather keep it simple though

Reply:
fwiw, i think it does make it simpler show the logic below. then it becomes below:
```
const [transform, routeFn] = routes[aEdge][bEdge]
info.apply(ElbowArrowTransform[transform])
const route = routeFn(info)
```
and you don't have to look at this mapping table to make sure that something is doing something custom (which it isn't). if there were special cases it'd be one thing, but this is consistent across all of them, so i think being terse here saves a little cognitive load!